specification/loadtestservice/resource-manager/Microsoft.LoadTestService/stable/2022-12-01/loadtestservice.json

```diff
@@ -0,0 +1,963 @@
+        }
+      }
+    },
+    "ResourceState": {
+      "description": "Resources provisioning states.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Deleted"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ResourceState",
+        "modelAsString": true
+      }
+    },
+    "QuotaResourceList": {
+      "description": "A list of Quota Bucket details. It contains an URL link to get the next set of results.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/QuotaResource"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "description": "List of Quota Bucket details by the loadtestservice resource provider",
+          "readOnly": true
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of Quota Bucket details results (if there are any).",
+          "readOnly": true
+        }
+      }
+    },
+    "QuotaResource": {
+      "type": "object",
+      "description": "Quota Bucket details",
+      "properties": {
+        "properties": {
+          "description": "Quota resource properties",
+          "$ref": "#/definitions/QuotaResourceProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "QuotaResourceProperties": {
+      "description": "Quota resource properties.",
+      "type": "object",
+      "properties": {
+        "limit": {
+          "description": "Quota Limit of the quota bucket.",
+          "type": "integer",
+          "minimum": 0,
+          "format": "int32"
+        },
+        "usage": {
+          "description": "Current Usage of the quota bucket.",
+          "type": "integer",
+          "minimum": 0,
+          "format": "int32"
+        },
+        "provisioningState": {
+          "description": "Resource provisioning state.",
+          "$ref": "#/definitions/ResourceState",
+          "readOnly": true
+        }
+      }
+    },
+    "CheckQuotaAvailabilityResponse": {
+      "description": "Check Quota Availability Response.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Check Quota Availability Response properties",
+          "$ref": "#/definitions/CheckQuotaAvailabilityResponseProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "CheckQuotaAvailabilityResponseProperties": {
+      "description": "Check Quota Availability Response properties.",
+      "type": "object",
+      "properties": {
+        "isAvailable": {
```

Comment:
>isAvailable [](http://example.com/codeflow?start=9&length=11)

ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive, flexible and future-proof while being easier for customers to discover, understand and use. Ref: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review). They also allow adding values later without requiring a new API version (e.g. "Unknown", "Default", etc.), and allow values other than just true/false. Even if you only ever want two values, you can have a string enum with True/False (or Enabled/Disabled, etc.).

In this case, you could consider availability: { Available, Unavailable }, or isAvailable: { Yes, No }, or something else you think of that's better.